sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/DefaultStep.java

```diff
@@ -139,6 +139,9 @@ public String getName() {
     @Override
     public Status getStatus() {
         synchronized (statusLock) {
+            if (status == Status.PENDING && getStrategy().isInterrupted()) {
+                return Status.WAITING;
```

Comment:
WAITING is a return value only. If you set status to WAITING, then you need to store previous value to revert back.  We have two variables(one is "status", and the other one is "interrupted"). "interrupted" is in strategy, so in all plan, step, phase.  "status" is in DefaultStep only. There is no setStatus in Element interface.